hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/index/IndexComponentFilterResolver.java

```diff
@@ -142,9 +149,14 @@ private static IndexComponentFilter convertFromRangeFilters(
                 toInclusive = candidateFilter.isToInclusive();
                 expressions.add(candidate.getExpression());
             }
+
+            if (from == null && to == null && candidateFilter.getFrom() == null && candidateFilter.getTo() == null) {
```

Comment:
Not related to this PR, but feels like all these specific condition checks to handle NULL should live somewhere outside of this class or alternatively be moved to their own separate methods to convey logic more clearly, although maybe comments would work too.

Reply:
the null/NULL checks generally feel very inconsitent, especially in different layers (optimizer, DAG, Index and IndexStore, Predicate API has also its own logic). I had lot's of headaches because of that but I did not feel confident enough to make it more consistent. Sometimes null means NULL, sometimes unbounded (including NULL or not), sometimes it is invalid... (probably somewhere is also means NOT NULL - for range filters)